openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,864 @@
-            name = json[j].mapped_value;
-            cat = json[j].category;
-            type = json[j].type;
-            grids = json[j].grids;
-            pDef = json[j].project_definition;
-
-            if (cat != undefined && type == 'svir-qgis') {
-                layerId = json[j].id;
-                layerTitle = json[j].mapped_value;
-                layerNames[name].push(layerId);
-                layersByCat[cat].push(layerTitle);
-                projectDefinition[name].push(pDef);
+    }
+    return primaryRiskIndicatorObj;
+}
+
+function createIndex(la, index) {
+    var ct = 0;
+    var indicator = [];
+    // setup the indicator with all the municipalities
+    for (var ia = 0; ia < la.length; ia++) {
+        var temp = {};
+        temp.region = la[ia].properties[selectedRegion];
+        indicator.push(temp);
+    }
+    for (var i = 0; i < index.length; i++) {
+        for (var j = 0; j < la.length; j++, ct++) {
+            if (indicator[j].region == la[j].properties[selectedRegion]) {
+                var tempName = index[i].name;
+                var tempValue = la[j].properties[tempName];
+                indicator[j][tempName] = tempValue;
             }
         }
-
-        // Get unique category names
-        var categoryUnique = categoryList.filter(function(itm,i,categoryList){
-            return i==categoryList.indexOf(itm);
-        });
-
-        for (var l in categoryUnique) {
-            // Append category names to dropdown list
-            var categoryTitle = categoryUnique[l];
-            var opt = document.createElement('option');
-            opt.innerHTML = categoryTitle;
-            opt.value = categoryTitle;
-            selCat.appendChild(opt);
-            // Append layer list to dowpdown
-            var layerOpt = document.createElement('option');
+    }
+    return indicator;
+}
+
+function combineIndicators(nameLookUp, themeObj, JSONthemes) {
+    projectDef = sessionProjectDef;
+    var ct = 0;
+    var subIndex = {};
+    var operator;
+    for (var y = 0; y < projectDef.children.length; y++) {
+        if (projectDef.children[y].name == nameLookUp) {
+            operator = projectDef.children[y].operator;
         }
-
-    });
-
-    // Create dynamic categorized layer dialog
-    $('#layer-category').change(function() {
-        // Remove the layer list element
-        document.getElementById('layer-list').options.length = 0;
-
-       // Create the layer list ibased on the category selected
-        var e = document.getElementById('layer-category');
-        var strUser = e.options[e.selectedIndex].value;
-        var layersArray = layersByCat[strUser];
-
-        for (var i in layersArray) {
-            var layers = layersArray[i];
-            var opt = document.createElement('option');
-            opt.innerHTML = layers;
-            opt.valuse = layers;
-            selLayer.appendChild(opt);
+    }
+    // first create an object with all of the district names
```

Comment:
I think all the occurrences of `district` and `region` should be refactored into the more generic `zone`